flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,446 @@
+    ) -> Response:
+        """
+        Send a request to Flood.
+        Returns None if the response status_code is not in the success_list.
+        """
+
+        try:
+            response = task.requests.request(
+                method, urllib.parse.urljoin(config["url"], url), **kwargs
+            )
+        except RequestException as e:
+            raise PluginError(
+                f"An error occurred while attempting to send a request to Flood: {e}"
+            )
+
+        if not response.status_code in success_list:
+            raise PluginError(
+                f"Failed to send request to Flood. Received status {response.status_code} expected one of {success_list}."
+            )
+
+        return response
+
+    @staticmethod
+    def is_jwt_expired(task: Task) -> bool:
+        """
+        Check if the JWT token is expired.
+        Returns True if the JWT token is expired or if the JWT token is not set.
+        """
+
+        jwt = task.requests.cookies.get('jwt')
+
+        if not jwt:
+            return True
+
+        try:
+            decoded_jwt = decode(jwt, algorithms="HS256", options={"verify_signature": False})
+        except:
+            return True
+
+        return decoded_jwt.get("exp", 0) <= time.time()
+
+    @staticmethod
+    def authenticate(task: Task, config: dict) -> None:
+        """
+        Authenticate with Flood and obtain a JWT token.
+        """
+
+        # Attempt to authenticate with Flood.
+        FloodClient.request(
+            task,
+            config,
+            "post",
+            "api/auth/authenticate",
+            success_list=[200],
+            json={"username": config["username"], "password": config["password"]},
+        )
+
+        # Check if the JWT token is expired or unset.
+        if FloodClient.is_jwt_expired(task):
+            raise PluginError("Failed to authenticate with Flood. No valid JWT was found.")
+
+        logger.debug("Successfully authenticated with Flood.")
+
+    @staticmethod
+    def add_torrent_urls(
+        task: Task,
+        config: dict,
+        urls: list,
+        destination: str = None,
+        tags: list = None,
+        start: bool = True,
+    ) -> None:
+        """
+        Add a list of torrent URLs to Flood.
+        Will only attempt to add torrents if the JWT token is not expired or unset.
+        """
+
+        if not urls:
+            raise PluginError("Parameter 'urls' cannot be empty.")
+        if FloodClient.is_jwt_expired(task):
+            raise PluginError("Not authenticated with Flood.")
+
+        data = {"urls": urls, "start": start}
+
+        if destination:
+            data["destination"] = destination
+        if tags:
+            data["tags"] = tags
+
+        # Attempt to add the torrent URLs to Flood.
+        response = FloodClient.request(
+            task, config, "post", "api/torrents/add-urls", success_list=[200, 202], json=data
```

Comment:
You could probably make the list of acceptable response codes a named `Final` variable at the top of the file for readability (and maybe a tiny performance boost).

Reply:
I guess the `[ 200, 202 ]` is acceptable under every scenario. I'll do that.